src/Compilers/Core/Portable/SourceGeneration/GeneratorAdaptor.cs

```diff
@@ -24,11 +24,6 @@ public SourceGeneratorAdaptor(ISourceGenerator generator)
 
         public void Initialize(IncrementalGeneratorInitializationContext context)
         {
-            // PROTOTYPE(source-generators):
-            // we'll call initialize on the underlying generator and pull out any info that is needed
-            // then we'll set PostInit etc on our context and register a pipleine that
-            // executes the old generator in the new framework
-
             GeneratorInitializationContext oldContext = new GeneratorInitializationContext(context.CancellationToken);
```

Comment:
Because then we wouldn't be able to share the semantic model across the generators. This is 'in' the graph, and to share state we need to be 'out' of the graph, so have to do the syntax walk before we get to this point.

Reply:
I actually had some ideas about how to make this simpler, but it'll need a bunch of re-work to the graph so didn't want to go down that path just yet.

Reply:
Gotcha.